psst-gui/src/webapi/client.rs

```diff
@@ -452,75 +808,247 @@ impl WebApi {
 
     // https://developer.spotify.com/documentation/web-api/reference/save-tracks-user/
     pub fn save_track(&self, id: &str) -> Result<(), Error> {
-        let request = self.put("v1/me/tracks")?.query("ids", id);
+        let request = self.put("v1/me/tracks", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 
     // https://developer.spotify.com/documentation/web-api/reference/remove-tracks-user/
     pub fn unsave_track(&self, id: &str) -> Result<(), Error> {
-        let request = self.delete("v1/me/tracks")?.query("ids", id);
+        let request = self.delete("v1/me/tracks", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 
     // https://developer.spotify.com/documentation/web-api/reference/save-shows-user
     pub fn save_show(&self, id: &str) -> Result<(), Error> {
-        let request = self.put("v1/me/shows")?.query("ids", id);
+        let request = self.put("v1/me/shows", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 
     // https://developer.spotify.com/documentation/web-api/reference/remove-shows-user
     pub fn unsave_show(&self, id: &str) -> Result<(), Error> {
-        let request = self.delete("v1/me/shows")?.query("ids", id);
+        let request = self.delete("v1/me/shows", None)?.query("ids", id);
         self.send_empty_json(request)?;
         Ok(())
     }
 }
 
 /// View endpoints.
 impl WebApi {
-    pub fn get_made_for_you(&self) -> Result<Vector<Playlist>, Error> {
+    pub fn get_user_info(&self) -> Result<(String, String), Error> {
         #[derive(Deserialize)]
-        struct View {
-            content: Page<Playlist>,
+        struct User {
+            region: String,
+            timezone: String,
         }
-
+        let token = self.access_token()?;
         let request = self
-            .get("v1/views/made-for-x")?
-            .query("types", "playlist")
-            .query("limit", "20")
-            .query("offset", "0");
-        let result: View = self.load(request)?;
-        Ok(result.content.items)
+            .agent
+            .request("GET", &format!("http://{}/{}", "ip-api.com", "json"))
+            .query("fields", "260")
+            .set("Authorization", &format!("Bearer {}", &token));
+
+        let result: User = self.load(request)?;
+        
+        Ok((result.region, result.timezone))
+        
+    }
+
+    fn build_home_request(&self, section_uri: &str) -> (String, String) {
+        let extensions = json!({
+            "persistedQuery": {
+                "version": 1,
+                "sha256Hash": "4da53a78e4e98d4f3fa55698af5b751fe05ca3a1a4a526ff8147e8866ccfa49f"
+            }
+        });
```

Comment:
Where did this hash come from?

Reply:
From the spotube file: https://github.com/KRTirtho/spotube/blob/9b024120601c0d381edeab4460cb22f87149d0f8/lib%2Fservices%2Fcustom_spotify_endpoints%2Fspotify_endpoints.dart

But maybe we should re-get it! This is the website referenced by spotube  https://www.apollographql.com/docs/graphos/operations/persisted-queries/

Reply:
I'll revert it directly to the one from spotube, then we can keep an eye on theirs and up date it from there!